### PR TITLE
Add Spark

### PR DESCRIPTION
**Pull request by Jeff-CCH**
- [x] add Spark on Ethereum and Gnosis Chain
- [x] add getTokenList in Permit2
- [x] add multichain to logics
- [x] yarn changeset
- [x] test

### CODE REVIEW

**Review by chouandy — APPROVED** [review]
LGTM

File: src/logics/spark/service.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    const testCases = [
+      { assets: [mainnetTokens.WETH, mainnetTokens.USDC] },
+      { assets: [mainnetTokens.WBTC, mainnetTokens.USDT] },
+    ];
+
+    testCases.forEach(({ assets }, i) => {
+      it(`case ${i + 1}`, async function () {
+        const flashLoanConfiguration = await service.getFlashLoanConfiguration(assets);
+        console.log(flashLoanConfiguration);
```
Comment (zodahu) [inline src/logics/spark/service.test.ts:62]: remove debug log

File: src/logics/spark/tokens/data/mainnet.json
Comment (zodahu) [inline src/logics/spark/tokens/data/mainnet.json:1]: Can we copy `script/refresh-tokens-data.ts` from `aave-v3` to generate this file? It helps generate the same sequence of tokens by everyone.
image

File: src/logics/spark/logic.supply.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      tokenList.push([reserveToken.asset, reserveToken.aToken]);
+    }
+
+    return tokenList;
+  }
+
+  async quote(params: SupplyLogicParams) {
+    const { input, tokenOut } = params;
+    const output = new common.TokenAmount(tokenOut, input.amount);
+    return { input, output };
```
Comment (zodahu) [inline src/logics/spark/logic.supply.ts:40]: As this is copied from `aave-v3`, prefer to keep the same code format without changing empty lines, it helps to future code refactor such as adding `modules/aavev3` similar to `modules/univ3`.

image

File: src/logics/uniswap-v3/configs.ts [hunk 3]
```diff
@@ -42,6 +42,15 @@ export const configs: univ3.Config[] = [
     feeAmounts: [FeeAmount.LOWEST, FeeAmount.LOW, FeeAmount.MEDIUM, FeeAmount.HIGH],
     bases: [polygonTokens.WETH, polygonTokens.USDC, polygonTokens.DAI, polygonTokens.USDT, polygonTokens.WBTC],
   },
+  {
+    // https://github.com/Uniswap/interface/blob/v4.270.6/src/constants/routing.ts#L97
+    chainId: common.ChainId.base,
+    factoryAddress: '0x33128a8fC17869897dcE68Ed026d694621f6FDfD',
+    quoter: { address: '0x3d4e44Eb1374240CE5F1B871ab261CD16335B76a', isV2: true },
+    swapRouterAddress: '0x2626664c2603336E57B271c5C0b26F421741e481',
```
Comment (zodahu) [inline src/logics/uniswap-v3/configs.ts:50]: Isn't the one on the base also SwapRouter02?

Comment (Jeff-CCH) [reply src/logics/uniswap-v3/configs.ts:50]: woops. The name on the [deployment page](https://docs.base.org/contracts/) is swapRouter but the contract name is swapRouter02. Will remove this one.

Comment (zodahu) [reply src/logics/uniswap-v3/configs.ts]: I think this page is more accurate https://docs.uniswap.org/contracts/v3/reference/deployments

**Pull request merged**